### ISSUE TITLE
linter: typescript/no-namespace cannot be configured in .oxlintrc.json

### ISSUE THREAD

**Issue by jvrdnd** [oxc-project/oxc#20982]
### What version of Oxlint are you using?

1.58.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["typescript"],
  "categories": {
    "correctness": "error",
    "suspicious": "warn",
    "pedantic": "warn",
    "style": "warn",
    "perf": "warn",
    "restriction": "warn",
    "nursery": "warn"
  },
  "rules": {
    "typescript/no-namespace": "off"
  }
}
```

### What happened?

When you attempt to modify the rule in .oxlintrc (e.g., in my config to disable, or make it err), it does not modify the result (it is still reported as warn). If I modify a specific file with `// oxlint-disable no-namespace`, it correctly disables the rule.

**Comment by camc314** [comment]
Hi!

thanks for filing this issue.

I tried to reproduce this with the config above, but was unable to reproduce the expected behaviour.

Please take a look at my reproduction as it is possible I have missed something obvious.

I observed `typescript/no-namespace` being correctly disabled when specified in `rules` object - when I set the severity to `error`, it the procued the `error` severity for the diagnostic.

Apologies in the delay for getting an answer to this issue. I'm going to close it out as can't reproduce, but if you've got more context, please add it and I'll be happy to get back to addressing it!

Thanks.
